compiler/src/main/kotlin/edu/cornell/cs/apl/viaduct/backends/CodeGenerationBackend.kt

```diff
@@ -1,7 +1,24 @@
 package edu.cornell.cs.apl.viaduct.backends
 
+import edu.cornell.cs.apl.viaduct.backends.aby.ABYBackend
+import edu.cornell.cs.apl.viaduct.backends.aby.ABYProtocolFactory
 import edu.cornell.cs.apl.viaduct.backends.cleartext.CleartextBackend
 import edu.cornell.cs.apl.viaduct.backends.commitment.CommitmentBackend
+import edu.cornell.cs.apl.viaduct.selection.ProtocolFactory
+import edu.cornell.cs.apl.viaduct.selection.unions
+import edu.cornell.cs.apl.viaduct.syntax.intermediate.ProgramNode
 
 /** Combines all back ends that support code generation. */
-val CodeGenerationBackend: Backend = listOf(CleartextBackend, CommitmentBackend).unions()
+object CodeGenerationBackend : Backend by backends.unions() {
+    override fun protocolFactory(program: ProgramNode): ProtocolFactory {
+        // TODO: fix ABYProtocolFactory to get rid of this hack
```

Comment:
what is the fix for `ABYProtocolFactory`?

Reply:
That TODO is copied from code that I wrote. `ABYProtocolFactory` should not need a reference to the parent factory. I think we discussed a change that would make that possible, but I don't quite remember what it was.